examples/advanced.zig

```diff
@@ -112,6 +112,92 @@ fn postMutliPart(easy: Easy) !void {
     std.debug.print("resp:{s}\n", .{resp.body.?.items});
 }
 
+fn iterateHeaders(easy: Easy) !void {
+    // Reset old options, e.g. headers.
+    easy.reset();
+
+    const resp = try easy.get("https://httpbin.org/response-headers?X-Foo=Hello&X-Foo=World&x-fOO=42");
+    defer resp.deinit();
+
+    std.debug.print("Iterating all headers...\n", .{});
+    {
+        var iter = try resp.iterateHeaders(.{});
+        while (try iter.next()) |header| {
+            std.debug.print("  {s}: {s}\n", .{ header.name, header.get() });
+        }
+    }
+
+    std.debug.print("Iterating X-Foo only...\n", .{});
+    {
+        const Entry = struct {
+            name: []const u8,
+            value: []const u8,
+            visited: bool = false,
+        };
+        var entries = [_]Entry{
+            .{ .name = "X-Foo", .value = "Hello" },
+            .{ .name = "X-Foo", .value = "World" },
+            .{ .name = "x-fOO", .value = "42" },
+        };
+
+        var iter = try resp.iterateHeaders(.{ .name = "X-Foo" });
+        while (try iter.next()) |header| {
```

Comment:
I would do check like this:
```
var i:usize = 0;
while(..) |header| {
  try expectEqualDeep(header, expected[i]);
  i += 1;
}

expect(i, 3);
```

`visited` is not needed.

Reply:
But the order of headers are not guarenteed, is it?

Reply:
If the order is unclear, I think we can ignore this check. 

I want to keep the example simple.